addons/udes_transport_management/models/transport.py

```diff
@@ -59,29 +59,38 @@ def get_info(self, **kwargs):
         return res
 
     def prepare_transport_info(self, **kwargs):
-        """Prepare transport info method that can be override in order
+        """Prepare transport info method that can be overridden in order
         to be able to extend it in other modules"""
-        self.ensure_one()
-        transport_info = [
-            {
-                "label": "Trailer Number",
-                "value": self.vehicle_sequence or None,
-                "lookup_key": "trailer_number",
-            },
-            {
-                "label": "Trailer Unit ID",
-                "value": self.vehicle_description or None,
-                "lookup_key": "trailer_unit_id",
-            },
-            {
-                "label": "Vehicle Registration",
-                "value": self.license_plate or None,
-                "lookup_key": "vehicle_registration",
-            },
-            {
-                "label": "Driver Name",
-                "value": self.driver_name or None,
-                "lookup_key": "driver_name",
-            }
-        ]
-        return transport_info
+        blank_transport = (
+                {
+                    "label": _("Trailer Number"),
+                    "value": None,
+                    "lookup_key": "trailer_number",
+                },
+                {
+                    "label": _("Trailer Unit ID"),
+                    "value": None,
+                    "lookup_key": "trailer_unit_id",
+                },
+                {
+                    "label": _("Vehicle Registration"),
+                    "value": None,
+                    "lookup_key": "vehicle_registration",
+                },
+                {
+                    "label": _("Driver Name"),
+                    "value": None,
+                    "lookup_key": "driver_name",
+                },
+            )
+        result = []
+        for transport in self:
+            transport_info = blank_transport.copy()
+            transport_info["trailer_number"].value = transport.vehicle_sequence
+            transport_info["trailer_unit_id"].value = transport.vehicle_description
+            transport_info["vehicle_registration"].value = transport.license_plate
+            transport_info["driver_name"].value = transport.driver_name
+            result.append(transport_info)
+        if not len(result):
```

Comment:
nit: if not result?

Reply:
I just realised that's not really clear but it's basically saying If not 0 then. Since 0 would interpreted as false. Would you prefer me to make it more explicit?

Reply:
i think if not result would be ok, no need for len

Reply:
Agree, no need for len, "not []" is commonly used.